### PR TITLE
Add log endpoint in restful plugin

### PR DESCRIPTION
**Pull request by eeff**
1. Add get log json api
2. Add log handler

### CODE REVIEW

File: plugins/restful/log_handle.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                // outside offset range
+                break;
+            }
+            len1 += len2;
+        }
+        if (len2 < 0) {
+            // outside offset range
+            break;
+        }
+
```
Comment (fengzeroz) [inline plugins/restful/log_handle.c:187]: should be able to delete

Comment (eeff) [reply plugins/restful/log_handle.c:187]: what do you mean?

File: include/json/neu_json_log.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+    int64_t since;
+    int64_t until;
+    char *  level;
+} neu_json_get_log_req_t;
```
[the remark this answers is not in the document]
Comment (eeff) [reply include/json/neu_json_log.h:38]: fixed.

File: plugins/restful/log_handle.c [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+void handle_get_log(nng_aio *aio)
+{
+    REST_PROCESS_HTTP_REQUEST_VALIDATE_JWT(
+        aio, neu_json_get_log_req_t, neu_json_decode_get_log_req, {
+            neu_json_get_log_resp_t resp  = { 0 };
+            int                     level = NEU_LOG_TRACE;
+
+            if (req->since > req->until ||
+                0 != string_to_log_level(req->level, &level)) {
```
[the remark this answers is not in the document]
Comment (eeff) [reply plugins/restful/log_handle.c:62]: Add pagination with page size limit

File: plugins/restful/log_handle.c [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        }
+
+        uint64_t t = mktime(&tm);
+        if (ts <= t) {
+            hi = mid + len1;
+            continue;
+        } else {
+            lo = mid + len1;
+            continue;
+        }
```
[the remark this answers is not in the document]
Comment (eeff) [reply plugins/restful/log_handle.c:195]: fixed.

**Pull request merged**